server/src/main/java/org/apache/cassandra/sidecar/handlers/OpenApiHandler.java

```diff
@@ -0,0 +1,164 @@
+ */
+@Singleton
+public class OpenApiHandler implements Handler<RoutingContext>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenApiHandler.class);
+
+    @Override
+    public void handle(RoutingContext context)
+    {
+        try
+        {
+            // Determine format from request path or Accept header
+            boolean isYaml = isYamlRequest(context);
+            String openApiContent = loadGeneratedOpenApiSpec(isYaml);
+            String contentType = isYaml ? "application/yaml" : "application/json";
+            
+            context.response()
+                   .putHeader("Content-Type", contentType)
+                   .end(openApiContent);
+        }
+        catch (Exception e)
+        {
+            LOGGER.warn("Failed to load generated OpenAPI specification, falling back to basic config", e);
+            context.response().setStatusCode(HttpResponseStatus.NOT_FOUND.code())
+                   .putHeader("Content-Type", "application/json")
+                   .end();
+        }
+    }
+
+    /**
+     * Determines if the request is for YAML format based on path or Accept header
+     */
+    private boolean isYamlRequest(RoutingContext context)
+    {
+        String path = context.request().path();
+        if (path != null && path.endsWith(".yaml"))
+        {
+            return true;
+        }
+        
+        String acceptHeader = context.request().getHeader("Accept");
+        return acceptHeader != null && acceptHeader.contains("application/yaml");
+    }
+    
+    /**
+     * Loads the generated OpenAPI specification from resources or build output
+     */
+    private String loadGeneratedOpenApiSpec(boolean isYaml) throws IOException
+    {
+        // First try to load from classpath resources (for packaged deployments)
+        String content = loadFromClasspathResource(isYaml);
+        if (content != null)
+        {
+            return content;
+        }
+
+        // Fallback to file system paths (for development)
+        return loadFromFileSystem(isYaml);
+    }
+
+    /**
+     * Attempts to load the OpenAPI spec from classpath resources
+     */
+    private String loadFromClasspathResource(boolean isYaml) throws IOException
+    {
+        String fileName = isYaml ? "openapi.yaml" : "openapi.json";
+        String resourcePath = "/openapi/" + fileName;
+        
+        InputStream inputStream = getClass().getResourceAsStream(resourcePath);
+        if (inputStream == null)
+        {
+            return null;
+        }
+        
+        LOGGER.debug("Loading OpenAPI specification from classpath: {}", resourcePath);
+        try 
+        {
+            byte[] bytes = inputStream.readAllBytes();
+            String content = new String(bytes, StandardCharsets.UTF_8);
+            LOGGER.info("Loaded OpenAPI specification from classpath resource");
+            return content;
+        }
+        finally
+        {
+            inputStream.close();
```

Comment:
we should just use vertx filesystem operations here, it allows you to read from the resources as well